src/OrbitGl/TimelineTicks.cpp

```diff
@@ -69,6 +69,23 @@ std::vector<uint64_t> TimelineTicks::GetMajorTicks(uint64_t start_ns, uint64_t e
   return major_ticks;
 }
 
+int TimelineTicks::GetTimestampsNumDigitsPrecision(uint64_t start_ns, uint64_t end_ns) const {
```

Comment:
Nit: From a testing perspective this is not ideal because you can't test the algorithm that determines the number of needed digits indepdendently from `GetMajorTicks`.

I would suggest to create a separate function that takes a single timestamp and calculates the number of decimal places needed.
Then I would implement another free function that takes a list (span) of timestamps and calculates the maximum number of decimal places needed. (Maybe this is overkill).
Afterwards this member function would only call GetMajorTicks and these other free functions.

Reply:
I would argue that this isn't relevant from a testing perspective. You're testing the public API of the class, and you don't really care if there is a dependency between them. Worst case, you're implicitly testing `GetMajorTicks` twice, but does that matter? You care about the outcome of this method when testing it, not how it's implemented.

Reply:
Well it makes the test more complex and harder to understand. I can't verify the result in my head without looking at the details of GetMajorTicks, GetAllTicks, and so on. I believe that's not great.

Reply:
Agree with both. The method is too complicated. My first plan was very different, harder to understand and didn't make sense to include only one timestamp. Now that I'm getting the number of digits, why not having a method who only return the precision of 1 timestamp? 

I will split it between two methods, the basic one belongs here and the other one might be also here (as a free function), or could be outside this class since I will only use it once. I think I will choose this last option.

Reply:
I've chosen the last option, and I will search for the maximum directly in TimelineUi component.